src/logging/logging.ts

```diff
@@ -5,37 +5,29 @@ import freedomTypes = require('freedom.types');
 
 // Perform log message formatting. Formats an array of arguments to a
 // single string.
-// TODO: switch to rest arguments.
 // TODO: move this into the provider.
-function formatStringMessageWithArgs_(msg:string, args?:any[])
+function formatStringMessageWithArgs_(args :any[])
     : string {
-  var formatted_msg = msg;
-  if (args && args.length) {
-    for (var i = 0; i < args.length; i++) {
-      formatted_msg = formatted_msg.replace('%' + (i + 1), args[i]);
-    }
-  }
-  return formatted_msg;
-}
+  var msg = '';
+  var arg :any;
 
-interface loggable {
-  (logger: freedomTypes.Logger): void;
-}
+  for (var i = 0; i < args.length; i++) {
+    arg = args[i];
+    if ('string' !== typeof(arg)) {
```

Comment:
I think you also need to consider the case where `arg` has been created via `new String`.